tests/integration_tests/conftest.py

```diff
@@ -14,3 +19,22 @@ def delete_bucket(bucket: str):
         objects = list(map(lambda x: {"Key": x["Key"]}, objects))
         _S3_CLIENT.delete_objects(Bucket=bucket, Delete={"Objects": objects})
     _S3_CLIENT.delete_bucket(Bucket=bucket)
+
+
+@pytest.fixture()
+def test_buckets():
+    """Sets up and tears down buckets used by this module.
+    On set up:
+    - creates test-bucket
+    - creates data-store-failed-files-unit-tests
+    - creates data-store-successful-files-unit-tests
+
+    On tear down, deletes all objects stored in the buckets and then the buckets themselves.
+    """
+    create_bucket(TEST_BUCKET)
+    create_bucket(Config.AWS_S3_BUCKET_FAILED_FILES)
+    create_bucket(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES)
+    yield
+    delete_bucket(TEST_BUCKET)
+    delete_bucket(Config.AWS_S3_BUCKET_FAILED_FILES)
+    delete_bucket(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES)
```

Comment:
Another thought I had is that at the moment this test fixture is at function scope so will setup and tear down new buckets for every test. I can't see that it's slowing down the tests (integration tests seem to take between 1m10s and 1m40s with either setting) so maybe this is fine as it is and has possible benefits like making it easier to debug bucket contents.

But perhaps there is an argument to be made for scoping this at session level and just doing setup and tear down once.

Reply:
Thanks for these comments, all make sense! I did look at making `test_buckets` module or session scoped, but thought the test creep of having certain files added to buckets from earlier tests could potentially lead to issues, but will test with session scope and make sure things are working as expected.

Reply:
@cyrusdobbs have just updated the PR with these suggested changes. I set the fixture scope for the `test_buckets` fixture to module which seemed like a reasonable compromise, so if test creep does occur it'll at least be from within the same module rather than from anywhere else.

Reply:
Makes sense, I think that's a nice middle ground